src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -142278,6 +142278,43 @@ static void F(object x, object? y)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "y").WithArguments("y", "object A.this[object? x, object y]").WithLocation(7, 22));
         }
 
+        [Fact]
+        [WorkItem(49754, "https://github.com/dotnet/roslyn/issues/49754")]
+        public void Issue49754()
+        {
+            var source0 =
+@"
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace Repro
+{
+	public class Test
+	{
+		public void Test(IEnumerable<(int test, int score)> scores)
+		{
+			scores.Select(s => (s.test, s.score switch
+			{
+
+			}));
+		}
+	}
+}"
+;
+            var comp = CreateCompilation(source0);
+            comp.VerifyEmitDiagnostics(
```

Comment:
>VerifyEmitDiagnostics [](start = 17, length = 21)

Do we call `ToPublicAnnotation()` here?

Reply:
> Do we call ToPublicAnnotation() here?

Yes, while performing nullable analysis for the code

---
In reply to: [538670812](https://github.com/dotnet/roslyn/pull/49849#discussion_r538670812) [](ancestors = 538670812)

Reply:
See this code in the walker:
```
                _analyzedNullabilityMapOpt[expr] = (new NullabilityInfo(result.LValueType.ToPublicAnnotation(), result.RValueType.State.ToPublicFlowState()),
                                                    // [https://github.com/dotnet/roslyn/issues/35046](https://github.com/dotnet/roslyn/issues/35046) We're dropping the result if the type doesn't match up completely
                                                    // with the existing type
                                                    expr.Type?.Equals(result.RValueType.Type, TypeCompareKind.AllIgnoreOptions) == true ? result.RValueType.Type : expr.Type);
```



---
In reply to: [538683287](https://github.com/dotnet/roslyn/pull/49849#discussion_r538683287) [](ancestors = 538683287,538670812)